### ISSUE TITLE
Second Cronworkflow is not being executed if first cronworkflow has 60k child workflow objects

### ISSUE THREAD

**Issue by sarabala1979** [argoproj/argo-workflows#4211]
## Summary 
The current Cronworkflow controller has only one worker to process the cronworkflow.  
it should be configurable like workflow workers.

What happened/what you expected to happen?

## Diagnostics

What Kubernetes provider are you using? 

What version of Argo Workflows are you running? 

```yaml
Paste a workflow that reproduces the bug, including status:
kubectl get wf -o yaml ${workflow} 
```

```
Paste the logs from the workflow controller:
kubectl logs -n argo $(kubectl get pods -l app=workflow-controller -n argo -o name) | grep ${workflow}
```

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by simster7** [comment]
Steps to reproduce?

**Comment by simster7** [comment]
> The current Cronworkflow controller has only one worker to process the cronworkflow.

It actually has two workers, but I agree it should be increased

**Comment by sarabala1979** [comment]
> > The current Cronworkflow controller has only one worker to process the cronworkflow.
> 
> It actually has two workers, but I agree it should be increased

Oh yes, I thought It is a single-threaded.

We observed the below scenario when we are debugging one of the EKS clusters last Friday.
1. Argo namespace has canary cronworkflow with 60k child workflow objects. 
2. We suspended canary workflow.
3. We created a new cronworkflow `canary-debug-session`
4. We didn't see a new cronworkflow triggered the workflows. At the same time, the EKS team was restarting the master. this may cause the informer to add a new object(not sure).
5. `canary-debug-session` was start working once all canary child workflows were deleted.
